### ISSUE TITLE
Upgrade fails if `tmp` is on different filesystem

### ISSUE THREAD

**Issue by sebdanielsson** [cupcakearmy/autorestic#191]
**Describe the bug**
`autorestic upgrade` fails with errors like:

* `os.Rename() failed (rename /tmp/autorestic-1252517049 /usr/local/bin/autorestic: invalid cross-device link), retrying with io.Copy()`
* `Error: open /usr/local/bin/autorestic: text file busy`. 

Se detailed output below.

**Expected behavior**
autorestic should upgrade itself.

**Environment**
 - OS: Fedora
 - Version: 35

**Additional context**
``` sh
❯ autorestic upgrade --verbose
Using config paths: . /root /root/.config/autorestic
restic already installed
> Executing: /usr/bin/restic self-update
writing restic to /usr/bin/restic
find latest release of restic at GitHub
restic is up to date

Downloading: https://github.com/cupcakearmy/autorestic/releases/download/v1.7.1/autorestic_1.7.1_linux_amd64.bz2
os.Rename() failed (rename /tmp/autorestic-1252517049 /usr/local/bin/autorestic: invalid cross-device link), retrying with io.Copy()
Error: open /usr/local/bin/autorestic: text file busy
cannot lock before reading config location
```

**Comment by cupcakearmy** [comment]
Can you reproduce it? Seemed fine to me on macos and linux

**Comment by cupcakearmy** [comment]
You probably have `/tmp` on a different file system I suppose?

https://stackoverflow.com/a/42400063/2425183

**Comment by SebDanielsson** [comment]
Not sure how to know. My fstab only contains an ext4 partition mounted at / and a swap partition.

**Comment by SebDanielsson** [comment]
It might be on snorher file system. Both Fedora and Arch use tmpfs.

https://wiki.archlinux.org/title/tmpfs#Usage

"Arch uses a tmpfs /run directory, with /var/run and /var/lock simply existing as symlinks for compatibility. It is also used for /tmp by the default systemd setup and does not require an entry in fstab unless a specific configuration is needed."

Not sure how to check if this is true for me but after some googling I found out that Fedora has been using tmpfs since version 18.

**Comment by SebDanielsson** [comment]
Just ran autorestic upgrade on my Manjaro machine and it said upgrade was successful but it basically just deleted itself. Trying to run the command again returns: command not found

**Comment by cupcakearmy** [comment]
That sounds bad, definitely need to look at the upgrade function.

**Comment by SebDanielsson** [comment]
Been playing around with datadog the last couple of days and discovered that Fedora is indeed using tmpfs so that is probably the issue☺️

**Comment by cupcakearmy** [comment]
TLDR; we need to copy and delete instead of renaming, otherwise it fails over different filesystems https://stackoverflow.com/a/42400063/2425183
